packages/context/test/acceptance/extension-point.ts

```diff
@@ -0,0 +1,215 @@
+  extensions: {
+    [extensionName: string]: any;
+  };
+  extensionPoint: {
+    [property: string]: any;
+  };
+}
+
+/**
+ * Base class for extension points
+ */
+export abstract class ExtensionPoint<EXT extends object> {
+  /**
+   * Configuration (typically to be injected)
+   */
+  protected configuration: ExtensionPointConfig;
+
+  constructor(
+    /**
+     * The unique name of this extension point. It also serves as the binding
+     * key prefix for bound extensions
+     */
+    protected name: string,
+    /**
+     * The Context (typically to be injected)
+     */
+    protected context: Context,
+    /**
+     * Configuration (typically to be injected)
+     */
+    configuration?: ExtensionPointConfig,
+  ) {
+    this.configuration = configuration || {extensionPoint: {}, extensions: {}};
+  }
+
+  /**
+   * Find an array of bindings for extensions
+   */
+  getExtensionBindings(): Binding[] {
+    return this.context.find(`${this.name}.*`);
+  }
+
+  /**
+   * Get a map of extension bindings by the keys
+   */
+  getExtensionBindingMap(): {[name: string]: Binding} {
+    const extensions: {[name: string]: Binding} = {};
+    const bindings = this.context.find(`${this.name}.*`);
+    bindings.forEach(binding => {
+      extensions[binding.key] = binding;
+    });
+    return extensions;
+  }
+
+  /**
+   * Look up an extension binding by name
+   * @param name Name of the extension
+   */
+  getExtensionBinding(name: string): Binding {
+    const key = `${this.name}.${name}`;
+    const extensions = this.getExtensionBindingMap();
+    return extensions[key];
+  }
+
+  /**
+   * Get an instance of an extension by name
+   * @param name Name of the extension
+   */
+  async getExtension(name: string): Promise<EXT> {
+    const binding = this.getExtensionBinding(name);
+    // Create a child context to bind `config`
+    const extensionContext = new Context(this.context);
+    extensionContext.bind('config').to(this.configuration.extensions[name]);
+    return await binding.getValue(extensionContext);
+  }
+}
+
+describe('Extension point', () => {
+  let ctx: Context;
+  beforeEach('given a context', createContext);
+
+  interface AuthenticationStrategy {
+    authenticate(credentials: any): Promise<boolean>;
+    config: any;
+  }
+
+  class LocalStrategy implements AuthenticationStrategy {
+    constructor(@inject('config') public config: any) {}
+
+    authenticate(credentials: any) {
+      return Promise.resolve(true);
+    }
+  }
+
+  class LDAPStrategy implements AuthenticationStrategy {
+    constructor(@inject('config') public config: any) {}
+
+    authenticate(credentials: any) {
+      return Promise.resolve(true);
+    }
+  }
+
+  class OAuth2Strategy implements AuthenticationStrategy {
+    constructor(@inject('config') public config: any) {}
+
+    authenticate(credentials: any) {
+      return Promise.resolve(true);
+    }
+  }
+
+  class AuthenticationManager extends ExtensionPoint<AuthenticationStrategy> {
```

Comment:
Using the name`Manager` is considered as antipattern, see e.g [here](https://stackoverflow.com/questions/1866794/naming-classes-how-to-avoid-calling-everything-a-whatevermanager) and [here](https://stackoverflow.com/questions/5569666/what-is-a-better-name-than-manager-processor-etc).

Let's use a more descriptive name please, for example `AuthenticationHandler`.